### PR TITLE
Update xmlhttprequest timeout and percentage complete

### PR DESCRIPTION
**Pull request by rotosonic**
Why: Edge 15 is periodically throwing error at the end of testing due to timing issue and need to report percentage complete to client
How: Update timing logic to remove error and return percentage complete
Test: check locally and run in edge 15 and verify test completes with no error

### CODE REVIEW

**Comment by Pradhyumna02** [comment]
forgot to make changes in downloadApp.js and uploadApp.js

File: public/lib/downloadHttpConcurrentProgress.js [hunk 1]
```diff
@@ -77,11 +79,12 @@
...
      */
     downloadHttpConcurrentProgress.prototype.onTestError = function (result) {
-        if (this._running) {
-            this.clientCallbackError(result);
-            clearInterval(this.interval);
-            this._running = false;
-        }
+      console.dir(result);
+      if (this._running) {
+         if ((Date.now() - this._beginTime) > this.testLength) {
```
Comment (Pradhyumna02) [inline public/lib/downloadHttpConcurrentProgress.js]: we might need a 'else' case here to send the error back if it is during the 12 sec

File: public/lib/downloadHttpConcurrentProgress.js [hunk 2]
```diff
@@ -77,11 +79,12 @@
      * @return error object
      */
     downloadHttpConcurrentProgress.prototype.onTestError = function (result) {
-        if (this._running) {
-            this.clientCallbackError(result);
-            clearInterval(this.interval);
-            this._running = false;
-        }
+      console.dir(result);
```
Comment (Pradhyumna02) [inline public/lib/downloadHttpConcurrentProgress.js]: a quick question: will it be any help to us having the console here.. without have it in the splunk?

File: public/lib/downloadHttpConcurrentProgress.js (hunk 1 above)
Comment (rotosonic) [inline public/lib/downloadHttpConcurrentProgress.js]: abort all requests and show opps

File: public/lib/uploadHttpConcurrentProgress.js [hunk 3]
```diff
@@ -85,10 +87,11 @@
...
      */
     uploadHttpConcurrentProgress.prototype.onTestError = function (result) {
+      console.dir(result);
         if (this._running) {
-            this.clientCallbackError(result);
-            clearInterval(this.interval);
-            this._running = false;
+          if ((Date.now() - this._beginTime) > this.testLength) {
+            this.endTest();
+          }
```
Comment (rotosonic) [inline public/lib/uploadHttpConcurrentProgress.js]: update similar to downloads

**Comment by abhi2687** [comment]
+1 once you address comment we discuss to add else block with abort all and Oops message

**Comment by kemarc** [comment]
+1 provisional, once you address comments

File: public/lib/downloadHttpConcurrentProgress.js (hunk 2 above)
Comment (rotosonic) [inline public/lib/downloadHttpConcurrentProgress.js]: yes for browser review and will pass to splunk dependent on test time

File: public/lib/downloadHttpConcurrentProgress.js (hunk 1 above)
Comment (rotosonic) [inline public/lib/downloadHttpConcurrentProgress.js]: updated

**Pull request merged**